File: backend/src/main/java/backend/entity/Product.java
package backend.entity;

import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;


@Getter
@Setter
public class Product {

    private Long id;

    private Category category;

    private String name;

    private String description;

    private Integer price;

    private String imageUrl;

    private Integer version = 0; // 商品情報の楽観ロック（必要に応じて使用）

    private Boolean deleted = false; // 論理削除フラグ

    private ProductStock productStock;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/types/product.ts
/**
 * 商品情報
 */
export interface Product {

    // 商品ID
    id: number;

    // 商品名
    name: string;

    //商品説明
    description: string;

    // 価格
    price: number;

    // 商品画像
    imageUrl: string | null;

    // カテゴリID
    categoryId: number;

    // 在庫数
    stockQuantity: number;

}